### PR TITLE
Release/2019-01-24

### PR DESCRIPTION
**Pull request by fisx**
## API changes

  * Track inviters of team members (#566)
  * New partner role. (#569, #572, #573, #576, #579, #584, #577, #592)
  * App-level websocket pongs. (#561)

## Bug fixes

  * Spar re-login deleted sso users; fix handling of brig errors. (#588)
  * Gundeck: lost push notifications with push-all enabled.  (#554)
  * Gundeck: do not push natively to devices if they are not on the whitelist.  (#554)
  * Gundeck: link gundeck unit tests with -threaded.  (#554)

## Internal changes

  * Get rid of async-pool (unliftio now provides the same functionality) (#568)
  * Fix: log multi-line error messages on one line. (#595)
  * Whitelist all wire.com email addresses (#578)
  * SCIM -> Scim (#581)
  * Changes to make the demo runnable from Docker (#571)
  * Feature/docker image consistency (#570)
  * add a readme, for how to build libzauth. (#591)
  * better support debian style machines of different architecturs (#582, #587, #583, #585, #590, #580)

### CODE REVIEW

**Comment by fisx** [comment]
> Current develop doesn't pass all tests.

Which tests fail?  I claim that's a CI issue and not one of the develop branch: I have seen multiple failures in may PRs these days of tests that pass fine locally.  But I may be wrong.

**Comment by jschaul** [comment]
```
Brig API Integration
  team
    invitation
      post /teams/:tid/invitations - 201:                  OK (0.72s)
      post /teams/:tid/invitations - 403 no permission:    OK (0.76s)
      post /teams/:tid/invitations - 403 too many pending: OK (1.12s)
      post /teams/:tid/invitations - roles:                OK (1.05s)
      post /register - 201 accepted:                       FAIL (0.72s)
        test/integration/API/Team.hs:204:
        Member has no/wrong invitation metadata
        expected: Just (87bc57e2-a14d-471e-a4a3-66b8ac62393e,2019-01-24T20:20:03.274Z)
         but got: Nothing
```

This is locally.

**Comment by neongreen** [comment]
@jschaul 

* We can't reproduce the test failure. (How likely/unlikely is it that you were running the test with an older version of Galley? That would expect it.)

* We're not doing a release on Friday; Tiago said he wants to look at it first, so Monday at the earliest.

File: libs/galley-types/src/Galley/Types/Teams.hs [hunk 1]
```diff
@@ -258,9 +309,17 @@ newTeam tid uid nme ico bnd = Team tid uid nme ico Nothing bnd
 newTeamList :: [Team] -> Bool -> TeamList
 newTeamList = TeamList
 
-newTeamMember :: UserId -> Permissions -> TeamMember
+newTeamMember :: UserId -> Permissions -> Maybe (UserId, UTCTimeMillis) -> TeamMember
 newTeamMember = TeamMember
 
+-- | This is called in "Galley.Data".  It throws an exception if the input is inconsistent, meaning
+-- that one of inviter and invitation timestamp is Nothing and the other is Just.  This is justified
+-- because it can only be caused by
```
Comment (tiago-loureiro) [inline libs/galley-types/src/Galley/Types/Teams.hs]: I guess this comment was unfinished :)

**Pull request merged**